Sync v3-0-stable with v3-0-test to release 3.0.4

jedcunningham opened the pull request:
This PR Syncs v3-0-stable with v3-0-test to release 3.0.4.

Release notes and version bumps added in

https://github.com/apache/airflow/commit/65c448e75fcfa7f66f97c28df7062def993b3a7e
https://github.com/apache/airflow/commit/4ff5f3ac2dd6add48d621e79d107bd08844718d2

Lzzz666 committed:
[Backport v3-0-test] Add back dag parsing pre-import optimization (#50371) (#52698)

* Add back dag parsing pre-import optimization (#50371)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

* Apply suggestions from code review

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

github-actions[bot] committed:
[v3-0-test] Advanced auto classification for provider documentation (#52902) (#52925)

(cherry picked from commit 19961713421885c8306d3a51a4c3eaf5f27f6cb7)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

potiuk committed:
[v3-0-test] Workaround the need to use latest version on Lucas pre-commit's (#52931) (#52942)

We are still waiting for release of the Lucas's pre-commit after
the https://github.com/Lucas-C/pre-commit-hooks/pull/103 has
been merged - and for now we need to use `bleeding-edge` for
the repo.

Also upgrades zizmor and solves using "env." in shell commands
that might lead to security issues.
(cherry picked from commit 535d71d91f36d3d9859d32e4695f0e81a2696a4b)

github-actions[bot] committed:
[v3-0-test] Make the update_installer_and_pre_commit script more resilient to re matches (#52960) (#52961)

(cherry picked from commit 8e5c284913f550b5b583e06f400f4e761c87fbd5)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

potiuk committed:
Remove upper-binding for "python-requires" (#52980) (#52984)

Follow up after #52967 -> from later discussions it turned out that
it's not really the ~= that is wrong and ambiguous, but that just
upper-binding of Python version is generally considered as a bad
idea - and it's not Astral's view but it's general consensus that
upper-binding of "python-requires" is bad. Since ~= implies
upper-binding, simply replacing it with >= is likely the best option
we can choose.

(cherry picked from commit e9eb4813e45f9e10125ce0e5befe110512b5c5f7)

potiuk committed:
[v3-0-test] Remove remnants of ~= used in requires-python. (#52985) (#52987)

Follow up after #52980 - there are still few more places where
the ~= was used in requires-python.
(cherry picked from commit 3f6f1db510ca9f6641c0c1bca396a20c3f930d00)

github-actions[bot] committed:
[v3-0-test] docs: update public interface doc to reflect airflow.sdk and AIP-72 (#52197) (#53117)

* docs: update public interface doc to reflect airflow.sdk and AIP-72

- Added a note under "Using Airflow Public Interfaces" to recommend using `airflow.sdk` as the official interface from Airflow 3.0.
- Referenced AIP-72 and linked related documentation.
- Encouraged users to prefer REST API and Python Client for integrations.

* Update airflow-core/docs/public-airflow-interface.rst

Great



---------
(cherry picked from commit e142ab96a0ecbb953e9fb8ab6d17b4c2a7624aba)

Co-authored-by: N R Navaneet <156576749+nrnavaneet@users.noreply.github.com>
Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

kaxil committed:
Run Task failure callbacks on DAG Processor when task is externally killed (#53058) (#53143)

Until https://github.com/apache/airflow/issues/44354 is implemented, tasks killed externally or when supervisor process dies unexpectedly, users have no way of knowing this happened.

This has been a blocker for Airflow 3.0 adoption for some:

- https://github.com/apache/airflow/issues/44354
- https://apache-airflow.slack.com/archives/C07813CNKA8/p1751057525231389

https://github.com/apache/airflow/issues/44354 is more involved and we might not get to it for Airflow 3.1 -- so this is a good fix until then similar to how we run Dag Run callback.

(cherry-picked from a5211f2efd5ccc565cbc16baee6144dba09918bc)

amoghrajesh committed:
[v3-0-test] Unify connection not found exceptions between AF2 and AF3 (#52968) (#53093)

github-actions[bot] committed:
[v3-0-test] Move --no-binary flag to uv command line from pyproject.toml (#53179) (#53185)

When the flag is specified in pyproject toml, it also forces no
binary installation of xmlsec and lxml for local uv sync which might
fail if some system libraries are not installed, so it is better
to do it in the image by passing the right flags to installer
directly.
(cherry picked from commit 47bbe55355757953507b8ae64cd05f9380e16748)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

github-actions[bot] committed:
[v3-0-test] Fix package typo in build images (#53209) (#53210)

(cherry picked from commit 45cc0d48521ec360bc77a3ddbaa6f3dde8af1cfd)

Co-authored-by: GPK <gopidesupavan@gmail.com>

potiuk committed:
Install python from source again (#53150) (#53202)

* Draft: Build python from source

Builds python from source, also installs
golang from official distribution. Does
both of these for the ci image only.

* Updates path

* Adds version upgrade check for python version

Adds support for using the airflow api to fetch
the newest python patch version available for specific
major_minor pair

* Updated to use args in dockerfile for python

* Added support for golang upgrade

* Fixed go version sorting in pre_commit install

* Added github token usage and fixed version regex

Updated python fetch request during upgrade to use
github token and fixed the regex

* Updated dockerfile.ci file

* Added support for multiple python versions

Adds python version from global consts into
build args now for the docker ci.

* Updated python install

* Increases timeout for ci image build

(cherry picked from commit 5aec2d5ea87bf110e384187851acfb9a856da38f)

Co-authored-by: Aritra Basu <24430013+aritra24@users.noreply.github.com>

potiuk committed:
[v3-0-test] Bump uv to 0.7.20 (#53207) (#53220)

(cherry picked from commit 13c28dc1f8b3837aab8c98460708fa7196b514ca)

Co-authored-by: GPK <gopidesupavan@gmail.com>

potiuk committed:
[v3-0-test] Pass GITHUB_TOKEN to update checks (#53219) (#53221)

Last update check failed where it should not because it got the
rate limit failure when checking for python updates - and the
reason was it did not have GITHUB_TOKEN set.
(cherry picked from commit a24962c84ddbdb5887853cace240a0821b733e15)

github-actions[bot] committed:
[v3-0-test] Bring back quick image build check and increase timeout minutes (#53227) (#53230)

The #53212 changed the quick-image-build check to only run on
canary build, but this was not the intention - and the image started
to fail because of timeout minutes were too short after we added
python building from sources.

This PR fixes it "properly" - changes timeout minutes to be slightly
longer than the timeout (900 seconds) we specify in build command
and brings back building the image on regular PRs.
(cherry picked from commit 5579edd4729f74ebe95418d57e764edaa26ab66c)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

potiuk committed:
[v3-0-test] bump remaining uv versions (#53228) (#53232)

(cherry picked from commit 43d8c3b0a96e36254ff6664bb6e2e1bf49c60bfc)

Co-authored-by: Bugra Ozturk <bugraoz93@users.noreply.github.com>

github-actions[bot] committed:
[v3-0-test] fix(ci): bump python versions (#53229) (#53233)

(cherry picked from commit 234c7f0c5cfd8f3d8f8d0dc441efda881178f75c)

Co-authored-by: Bugra Ozturk <bugraoz93@users.noreply.github.com>

github-actions[bot] committed:
[v3-0-test] Dev: Fix KeyError in release issue content generation (#52751) (#52756)

When generating issue content for releases, some PRs were being filtered out
during processing (e.g., dependabot PRs, doc-only PRs) but still remained in
the `pull_requests` dictionary. This caused a `KeyError` when the Jinja2 template
tried to access `user_logins` for PRs that had no corresponding user data.

The fix ensures that `pr_list` variable only contains PRs that have corresponding entries
in the users dictionary, preventing the template from accessing undefined keys.

Fixes issue where `breeze release-management generate-issue-content-core`
command failed with "UndefinedError: dict object has no element <PR_NUMBER>".
(cherry picked from commit 19da28163042a2a4924d905cdb81c22857486c73)

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

github-actions[bot] committed:
[v3-0-test] Update mypy script with --warn-unused-ignores (#53240) (#53245)

* Update mypy script with --warn-unused-ignores

* Add doc information
(cherry picked from commit 1d7e0aed6ca9f2603d1e9e8d3de2d9b8759d6823)

Co-authored-by: GPK <gopidesupavan@gmail.com>

github-actions[bot] committed:
[v3-0-test] Fix mypy cmd (#53246) (#53247)

(cherry picked from commit 555ca215ae5c07dd1430da52083c4a3abb1370cc)

Co-authored-by: GPK <gopidesupavan@gmail.com>

jason810496 committed:
Resolve OOM When Reading Large Logs in Webserver (#49470) (#53167)

* Add note for new usage of LogMetadata

* Add _stream_parsed_lines_by_chunk

* Refactor _read_from_local/logs_server as return stream

* Refactor _interleave_logs with K-Way Merge

* Add _get_compatible_log_stream

* Refactor _read method to return stream with compatible interface

- Add compatible interface for executor, remote_logs
- Refactor skip log_pos with skip for each log source

* Refactor log_reader to adapt stream

* Fix _read_from_local open closed file error

* Refactor LogReader by yielding in batch

* Add ndjson header to get_log openapi schema

* Fix _add_log_from_parsed_log_streams_to_heap
- Add comparator for StructuredLogMessage
- Refactor parsed_log_streams from list to dict for removing empty logs

* Fix _interleave_logs dedupe logic
- should check the current logs with default timestamp

* Refactor test_log_handlers
- Fix events utils
- Add convert_list_to_stream, mock_parsed_logs_factory utils
- Fix the following test after refactoring FileTaskHandler
    - test_file_task_handler_when_ti_value_is_invalid
    - test_file_task_handler
    - test_file_task_handler_running
    - test_file_task_handler_rotate_size_limit
    - test__read_when_local
    - test__read_served_logs_checked_when_done_and_no_local_or_remote_logs
    - test_interleave_interleaves
    - test_interleave_logs_correct_ordering
    - test_interleave_logs_correct_dedupe
- Add new test for refactoring FileTaskHandler
    - test__stream_lines_by_chunk
    - test__log_stream_to_parsed_log_stream
    - test__sort_key
    - test__is_sort_key_with_default_timestamp
    - test__is_logs_stream_like
    - test__add_log_from_parsed_log_streams_to_heap

* Move test_log_handlers utils to test_common

* Fix unit/celery/log_handlers test

* Fix mypy-providers static check

* Fix _get_compatible_log_stream
- sequential yield instead of parallel yield for all log_stream

* Fix amazon task_handler test

* Fix wask task handler test

* Fix elasticsearch task handler test

* Fix opensearch task handler test

* Fix TaskLogReader buffer
- don't concat buffer with empty str, yield directly from buffer

* Fix test_log_reader

* Fix CloudWatchRemoteLogIO.read mypy

* Fix test_gcs_task_handler

* Fix core_api test_log

* Fix CloudWatchRemoteLogIO._event_to_str dt format

* Fix TestCloudRemoteLogIO.test_log_message

* Fix es/os task_hander convert_list_to_stream

* Fix compact tests

* Refactor es,os task handler for 3.0 compact

* Fix compat for RedisTaskHandler

* Fix ruff format for test_cloudwatch_task_handler after rebase

* Fix 2.10 compat TestCloudwatchTaskHandler

* Fix 3.0 compat test for celery, wasb

Fix wasb test, spelling

* Fix 3.0 compat test for gcs

* Fix 3.0 compat test for cloudwatch, s3

* Set get_log API default response format to JSON

* Remove "first_time_read" key in log metadata

* Remove "<source>_log_pos" key in log metadata

* Add LogStreamCounter for backward compatibility

* Remove "first_time_read" with backward "log_pos" for tests

- test_log_reader
- test_log_handlers
- test_cloudwatch_task_handler
- test_s3_task_handler
- celery test_log_handler
- test_gcs_task_handler
- test_wasb_task_handler
- fix redis_task_handler
- fix log_pos

* Fix RedisTaskHandler compatibility

* Fix chores in self review

- Fix typo in _read_from_logs_server
- Remove unused parameters in _stream_lines_by_chunk
- read_log_stream
    - Fix doc string by removing outdate note
    - Only add buffer for full_download
- Add test ndjson format for get_log API

* Fine-tune HEAP_DUMP_SIZE

* Replace get_compatible_output_log_stream with iter

* Remove buffer in log_reader

* Fix log_id not found compact for es_task_handler

* Fix review comments
- rename LogStreamCounter as LogStreamAccumulator
- simply for-yield with yield-from in log_reader
- add type annotation for LogStreamAccumulator

* Refactor LogStreamAccumulator._capture method
- use itertools.isslice to get chunk

* Fix type hint, joinedload for ti.dag_run after merge

* Replace _sort_key as _create_sort_key

* Add _flush_logs_out_of_heap common util

* Fix review nits

 - _is_logs_stream_like
    - add type annotation
    - reduce to 1 isinstance call
- construct log_streams in _get_compatible_log_stream inline
- use TypeDict for LogMetadata
- remove len(logs) to check empty
- revert typo of self.log_handler.read in log_reader
- log_stream_accumulator
    - refactor flush logic
    - make totoal_lines as property
    - make stream as property

* Fix mypy errors after merge

* Fix redis task handler test

* Refactor _capture logic in LogStreamAccumulator

* Add comments for ingore LogMetadata TypeDict

* Add comment for offset; Fix commet for LogMessages

* Refactor with from_iterable, islice

* Fix nits in test

- refactor structured_logs fixtures in TestLogStreamAccumulator
- use f-strign in test_file_task_handler
- assert actual value of _create_sort_key
- add details comments in test__add_log_from_parsed_log_streams_to_heap

* Refactor test_utils

* Add comment for lazy initialization

* Fix error handling for _stream_lines_by_chunk

* Fix mypy error after merge

* Fix final review nits

* Fix mypy error

(cherry picked from commit ee54fe93768e70da5bd2f0bb7398707181bee26a)

github-actions[bot] committed:
[v3-0-test] Fix broken link in contributors doc on providers (#53344) (#53347)

(cherry picked from commit a6efa53040c43dbf2903cb667116ed162916e041)

Co-authored-by: Zach <zach.gottesman@datadoghq.com>

github-actions[bot] committed:
[v3-0-test] Update doc string for BaseOperator (#53403) (#53404)

(cherry picked from commit 06065f9b2cc1ac5e3d38964eb4139f6d51ae9a12)

Co-authored-by: Elad Kalif <45845474+eladkal@users.noreply.github.com>

potiuk committed:
[v3-0-test] Fixing uv upgrade failures on main (#53406) (#53451)

(cherry picked from commit 6aeee1563378269cd7782f15a1ffb64a819b4993)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

github-actions[bot] committed:
[v3-0-test] fix: Make ``task_success_overtime`` configurable (#53342) (#53351)

(cherry picked from commit afb2e8ab119a7eecb1a01d4f419deb7681948d81)

Co-authored-by: Kacper Muda <mudakacper@gmail.com>

potiuk committed:
[v3-0-test] Add changelog for regenerated image for Airflow 3.0.3 (#53516) (#53522)

(cherry picked from commit 11a636146dd0ab4b49c223ff0bcfc878af45fbf1)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

github-actions[bot] committed:
[v3-0-test] Make GITHUB_TOKEN variable mandatory when updating installers (#53452) (#53457)

When running "update-installers-and-pre-commit" pre-commit, the script
is virtually guaranteed to fail with "rate limits" reached if you do
not use GITHUB_TOKEN. This change makes GITHUB_TOKEN variable mandatory
for the pre-commit, it prints the helpful information and prints URL
that allows to create such GITHUB_TOKEN very easily.
(cherry picked from commit aeb0d4cafa11ddc0d29bd7c64117195f8ac3768e)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

potiuk committed:
[v3-0-test] Add sql extras to pandas in the providers where sqlalchemy is needed (#53535) (#53539)

Without those extras pandas does not impose limitations on sqlalchemy
version used - but it is really an implicit dependency of pandas,
when it is used for sqlalchemy interactions.

This will drop constraint version of pandas for Airflow 3 to 2.1 and
it is really a conflicting dependency for Python 3.13 where only pandas
2.2.3 can be used but it requires sqlalchemy 2. This extra should
be added while migrating to sqlalchemy 2 is complete.
(cherry picked from commit ff084ed8ff27827e6496fbae520953836051fd29)

github-actions[bot] committed:
[v3-0-test] DagBag in file parsing used to collect dags in dag processing uses default value for safe_mode, which resolves to value in configuration (#52694) (#53507)

(cherry picked from commit e311c6bf990bb9ab91afe1620e62baf19866d493)

Co-authored-by: BBQing <33732539+BBQing@users.noreply.github.com>

github-actions[bot] committed:
[v3-0-test] Handle invalid execution API urls gracefully in supervisor (#53082) (#53518)

(cherry picked from commit e9e6ac27a4ba434fefc024b0b6430fb319c609e5)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

amoghrajesh committed:
Remove warning for `BaseOperator.executor` because its false (#53496) (#53519)

(cherry picked from commit 00194da372d1726794e806fb9954ee0d97939b83)

potiuk committed:
[v3-0-test] Upgdate to uv 0.8.0 (#53552) (#53564)

(cherry picked from commit b9cddf4740fd8134a29e714dc49b12f855ecc7e4)

jason810496 committed:
[v3-0-test] Remove gunicorn daemonize for api-server (#52860) (#53372)

* [v3-0-test] Remove gunicorn daemonize for api-server (#52860)

* Remove gunicorn daemonize for api-server

* Rename _CommonCLIUvicornTestClass

* Add test for api-server with daemon option

* Fix setproctitle on MacOS

* Replace api/access_logfile with api/log_config

* Fix [api/log_config] cli_config

* Remove logging and unsed option in api_server_command

- the logging before daemonization is not logging the correct PID so
  remove it
- should_setup_logging is default as False

* Fix test_run_command_daemon

* Add significant rst

(cherry picked from commit ab1ee72c2bedf7b5559f6860c4004e3888c78e54)

* Fix spelling wordlist

* Fix configuration

potiuk committed:
[v3-0-test] Include fixed navbar for anchor position calculation. (#52016) (#53581)

(cherry picked from commit d8086e635779cbaabd7aa02af5787008a207eeac)

Co-authored-by: Josef Šimánek <josef.simanek@gmail.com>

github-actions[bot] committed:
[v3-0-test] fix: accordian child contents to be visible when overflown (#53595) (#53602)

(cherry picked from commit 04dcc2326b9a06459bfae7d97e983587a0c4753f)

Co-authored-by: Naseem Shah <naseemshahvarikodan@gmail.com>

github-actions[bot] committed:
[v3-0-test] fix(systemd): Update systemd unit files and documentation for Airflow 3.0 (#52294) (#53609)

* docs: update public interface doc to reflect airflow.sdk and AIP-72

- Added a note under "Using Airflow Public Interfaces" to recommend using `airflow.sdk` as the official interface from Airflow 3.0.
- Referenced AIP-72 and linked related documentation.
- Encouraged users to prefer REST API and Python Client for integrations.

* Update airflow-core/docs/public-airflow-interface.rst

Great



* fix(systemd): Update systemd service files and docs for Airflow 3.0

- Updated systemd unit files to support Airflow 3.0 architecture.
- Replaced deprecated `webserver` with `api-server` in ExecStart.
- Added service units for new components: `triggerer`, `dag-processor`, and `api-server`.
- Adjusted ExecStart commands to support virtualenv-based deployments.
- Improved documentation (`run-with-systemd.rst`) to clarify service changes, virtualenv usage, and installation paths.
- Linked to the updated GitHub `scripts/systemd` directory for latest service files.

Tested Airflow 3.0 manually on macOS using virtualenv, confirming correct startup behavior for `scheduler`, `api-server`, and `triggerer`.

* fix(systemd): Update systemd service files and docs for Airflow 3.0

* Minor test failure fixes

---------
(cherry picked from commit 097ae4517d06d0fae1ae5890e206fffa7e64c148)

Co-authored-by: N R Navaneet <156576749+nrnavaneet@users.noreply.github.com>
Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

github-actions[bot] committed:
[v3-0-test] Add deprecation notice for using Connection from models (#53594) (#53621)

(cherry picked from commit 747a3e703aa46fb41297d052060122575a59e766)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

dstandish committed:
[v3-0-test] Skip empty dag run conf rows and set statement timeout (#50788) (#53619)

Don't bother depickling `{}` object in dag run conf

Also, set the statement timeout a little higher for xcom migration.


(cherry picked from commit f75b60740e59d47a223b350f506d9100e76c584d)

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

github-actions[bot] committed:
[v3-0-test] Ignore cursor rules from the git tree (#53658) (#53659)

(cherry picked from commit 70a29c14ce3a267368aee2075bc90bdf60e2d1e9)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

eladkal committed:
Fix incorrect reference to SimpleHttpOperator in 3.0.0 release notes (#53661)

potiuk committed:
[v3-0-test] Add GITHUB_TOKEN in test helm release job (#53674) (#53677)

(cherry picked from commit fb7eac2)

potiuk committed:
[v3-0-test] Add possibility to apply a commit with patch to docs when publishing (#53668) (#53678)

You can specify commit hash to apply when you build documentation.
This allows to regenerate past version of the documentation by
checking out the exact version tag that was used back then and
applying the commit hash with fixes.

This might help in fixing issues like
https://github.com/apache/airflow/issues/53646

(cherry picked from commit d6a51e9ac3c02724896858c9970752001e030031)

github-actions[bot] committed:
[v3-0-test] Improve validation of breeze workflow-run command (#53685) (#53688)

The workflow run command validation had few places where the
validation could be improved:

* when you do not specify packages, the workflow has no idea what
  to build, so the list of packages should not be empty

* when gh has an old version, some of the functionality might not
  work - we are checking gh version to check if it is old enough.
(cherry picked from commit 0dc9105f085906232fbeeb6c2c028bc5d90c607e)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

github-actions[bot] committed:
[v3-0-test] Fix typo in AIRFLOW_COMMIT(S) diagnostics in workflow_run (#53686) (#53689)

(cherry picked from commit 0c178848bf42dcb1a48435ddb09056ba9625daf9)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

rawwar committed:
use of debugpy to run airflow components via breeze (#51763)

* debug support in breeze

* fix celery env var

* Add debug options for Airflow components and update related constants

* Add debug options for Airflow components and update related constants

* refactor

* update documentation and add setup_vscode

* Remove section on overriding default debug ports in VSCode setup documentation

* fix base-ports

* fix base-ports

* update docs

* add webserver details in docs

* update docs

* update docs

* update docs

(cherry picked from commit 55d648ec094fc6f22a28f34bef0656a062705494)

github-actions[bot] committed:
[v3-0-test] CHG: cover v1, v2 (#53636) (#53709)

(cherry picked from commit 9d9924e1933a6639fc2d8b2635bf054ed341828c)

Co-authored-by: Kyungjun Lee <kyungjunlee.me@gmail.com>

potiuk committed:
[v3-0-test] Bump gitpython from 3.1.44 to 3.1.45 in /airflow-core (#53725) (#53731)

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.44 to 3.1.45.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.44...3.1.45)
(cherry picked from commit 1361afafbf966c186c5158bbc9d0ab157e04e2e1)



---
updated-dependencies:
- dependency-name: gitpython
  dependency-version: 3.1.45
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

potiuk committed:
[v3-0-test] Bump gitpython from 3.1.44 to 3.1.45 (#53724) (#53732)

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.44 to 3.1.45.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.44...3.1.45)
(cherry picked from commit aa2b533bb3eb5648f85c7f76ae867c4f1378b244)



---
updated-dependencies:
- dependency-name: gitpython
  dependency-version: 3.1.45
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

github-actions[bot] committed:
[v3-0-test] this makes "all" not show all dags (#53656) (#53672)

(cherry picked from commit 5db1b715f7dac87c543865d1aece135a70f1c719)

Co-authored-by: Zach Liu <zachliu@users.noreply.github.com>

github-actions[bot] committed:
[v3-0-test] Remove extra slash from endpoint URL (#53755) (#53764)

(cherry picked from commit 31db12826ee05c0b63cc1fcdb0f17c32afb6e46b)

Co-authored-by: Lui Pillmann <lui.pillmann@gmail.com>

jason810496 committed:
[v3-0-test] Respect `apps` flags for api_server_command (#52929) (#53775)

* Respect apps flags for api_server_command

* Refactor test_api_apps_env

(cherry picked from commit c0c41ff134044c565516842bb089d30e827dccab)

potiuk committed:
[v3-0-test] Better type handling to account for types-docutils bump (#53364) (#53778)

(cherry picked from commit 7654e1a49b074d5dc06dd1218369399430dfcfc9)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

potiuk committed:
[v3-0-test] Make branch tests more resilient to provider dependency changes (#53776) (#53777)

While we want to follow automated upgrades of dependencies and constraints
on vX-Y-test branches, in order to pull latest providers from PyPI,
this might have some undesired effects - for example some provider's
dependencies might break tests from importing or produce false
tests results if such tests depend on - for example - importing all
provider example dags. This happens for example for serialization
tests - which import all example dags to test that serialization
works for them.

Such tests should be simply skipped when target branch is different
than main.

This PR adds "skip_if_not_on_main" marker and uses it for two tests
that are currently failing on v3-0-test branch due to weaviate
dependency upgrades causing import error and some undesired warnings.
(cherry picked from commit 92909e5)

gopidesupavan committed:
sphinx: avoid repeated isinstance checks (#53367) (#53783)

Trying to fix failing mypy devel-common

Co-authored-by: Dev-iL <Gid.CTO+github@gmail.com>

potiuk committed:
[v3-0-test] Add cert used for apiserver to client (fix for self-signed) (#53574) (#53793)

* Add cert used for apiserver to client (fix for self-signed)

* Correct indentation, run pre-commit checks

* Add test for cert load

* Update config description for api ssl_cert

* Use mock instead and check error when non-existant CA is not found
(cherry picked from commit 6dba10134a0beff4656b344652d9a6f09d0648fd)

Co-authored-by: Daniel Dean <dd@danieldean.uk>

github-actions[bot] committed:
[v3-0-test] Add documentation for self-signed certificate (#53788) (#53792)

(cherry picked from commit f83fe1923d0192b58fb827ac7cd6043eea4f1389)

Co-authored-by: Daniel Dean <dd@danieldean.uk>

github-actions[bot] committed:
[v3-0-test] Add user facing docs for running a separate Task Execution API-server (#53789) (#53794)

(cherry picked from commit 97549587cdefb4ab9ea6c08dc8a30023e00ccb9f)

Co-authored-by: LIU ZHE YOU <68415893+jason810496@users.noreply.github.com>

github-actions[bot] committed:
[v3-0-test] Remove unnecessary group_by clause in event logs query (#53733) (#53807)

The group_by(Log.id) clause was redundant since Log.id is the primary key
and doesn't affect the query results. This simplifies the query without
changing functionality.

Fixes #53695
(cherry picked from commit 85d80cdecd25ce3d6d57e36a71825b8af32218fa)

Co-authored-by: xiaohuanlin <33860497+xiaohuanlin@users.noreply.github.com>

github-actions[bot] committed:
[v3-0-test] Unconditionally disable start_from_trigger (#53744) (#53750)

(cherry picked from commit 61862eea2956b43793a984804db272dbc9aa8861)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

github-actions[bot] committed:
[v3-0-test] Serialize nan and infinities to string (#53835) (#53844)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

gopidesupavan committed:
Make BaseOperator on_kill functionality work with TaskSDK (#53718) (#53832)

* Make BaseOperator on_kill functionality work with TaskSDK

* Fix static checks

* Resolve review comments

* Update task-sdk/tests/task_sdk/execution_time/test_supervisor.py



---------

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

github-actions[bot] committed:
[v3-0-test] Allow Remote logging providers to load connections from the API Server (#53719) (#53761)

Often remote logging is down using automatic instance profiles, but not
always. If you tried to configure a logger by a connection defined in the
metadata DB it would have not worked (it either caused the supervise job to
fail early, or to just behave as if the connection didn't exist, depending on
the hook's behaviour)

Unfortunately, the way of knowing what the default connection ID various hooks
use is not easily discoverable, at least not easily from the outside (we can't
look at `remote.hook` as for most log providers that would try to load the
connection, failing in the way we are trying to fix) so I updated the log
config module to keep track of what the default conn id is for the modern log
providers.

Once we have the connection ID we know (or at least have a good idea that
we've got the right one) we then pre-emptively check the secrets backends for
it, if not found there load it from the API server, and then either way. if we
find a connection we put it in the env variable so that it is available.

The reason we use this approach, is that are running in the supervisor process
itself, so SUPERVISOR_COMMS is not and cannot be set yet.
(cherry picked from commit e4fb6862c6b16f790fed1baa4af8882125c984c9)

Co-authored-by: Ash Berlin-Taylor <ash@apache.org>

pierrejeambrun committed:
Revert "[v3-0-test] Fix editing connection with sensitive extra field (#52403…" (#53889)

This reverts commit 7be9da37c2bd453bd85d15c60de3350a0bfdefc2.

github-actions[bot] committed:
[v3-0-test] Removed bold formatting for Public Interface for Airflow 3.0+ (#53955) (#53964)

(cherry picked from commit 8e1201c7713d5c677fa6f6d48bbd4f6903505f61)

Co-authored-by: Ankit Chaurasia <8670962+sunank200@users.noreply.github.com>

pierrejeambrun committed:
Handle unmodified sensitived fields when updating connections (#53943) (#53973)

* Fix redacted values editing

* Small improvements

* Small adjustments

* Update UI and fix some errors

* Address PR comments

(cherry picked from commit 0abcfdf0e61ebbaa6ae24883642439f40852c325)

github-actions[bot] committed:
[v3-0-test] Add type annotation for dev system tests (#53979) (#53996)

(cherry picked from commit 647b1b9380c4082e67d8a6aeaf41cbfe89175feb)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

github-actions[bot] committed:
[v3-0-test] Fix nth-child warning (#53982) (#54000)

(cherry picked from commit 2156fce0be580d36aaeb13839a230fd4aebfa334)

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>

github-actions[bot] committed:
[v3-0-test] Fix scheduler heartbeat timeout failures with ``DetachedInstanceError`` (#53838) (#53858)

(cherry picked from commit 94580537f56a2123f103640aaa6c87b4462c1b30)

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

potiuk committed:
fix(diagrams): update labels from "Webserver(s)" to "API Server(s)" in architecture diagrams (#53917) (#54020)

(cherry picked from commit 9383b0c0af4928fd8dfd6df729cd05b39ff8d2a8)

Co-authored-by: Aayush Bisen <aayushbisen@outlook.com>

Lee-W committed:
docs: Update DAG author as Dag author (#53857) (#53950)

potiuk committed:
[v3-0-test] build: update uv to 0.8.4 and pip to 25.2 (#53954) (#54021)

(cherry picked from commit c4bc393b088e59c0a231e8b87d3b88155cbfa0d8)

Co-authored-by: Wei Lee <weilee.rx@gmail.com>

jedcunningham committed:
Revert "[v3-0-test] Remove gunicorn daemonize for api-server (#52860) (#53372)" (#54002)

This reverts commit 7212c8876efaa64e2d62d320ab53ef85491dcd65.

pierrejeambrun committed:
Unmount dialog & add onOpen (#53307) (#54025)

(cherry picked from commit bb87ad5da725c530be06fc2bcf13c3c9d579496a)

Co-authored-by: Wei-Yu Chen <60685934+Jasperora@users.noreply.github.com>

jedcunningham committed:
Fix version for `parsing_pre_import_modules` config option move (#53995) (#54011)

This will land in 3.0.4, not 3.1.0.

(cherry picked from commit bbca500d712025429cde6e4ad792b7a52bf5ef6d)

github-actions[bot] committed:
[v3-0-test] Fix AIRFLOW_API_APPS constant in api_server_command (#54007) (#54012)

(cherry picked from commit d2449821512c81b828c5022d224b273f172384f6)

Co-authored-by: LIU ZHE YOU <68415893+jason810496@users.noreply.github.com>

jscheffl committed:
[v3-0-test] Flexible form use ReactMarkdown instead of default Markdown component (#54032) (#54040)

(cherry picked from commit 805a950365f99d86f5ba414843a17740b47d3c4f)

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>

jedcunningham committed:
Bump Airflow to 3.0.4 and Task SDK to 1.0.4

jedcunningham committed:
Add Release notes for Airflow 3.0.4 and Task SDK 1.0.4

potiuk reviewed (changes requested):
I mark it as request changes until we agree if there is a potential security issue to fix.

github-actions[bot] committed:
[v3-0-test] Fix `_extract_template_fields ` mypy error (#53855) (#54047)

(cherry picked from commit e809161694b4bd99c265b08f6f701904599653f8)

Co-authored-by: Yeonguk Choo <choo121600@gmail.com>

github-actions[bot] committed:
[v3-0-test] Update docs for Airflow CLI help command (#51042) (#53383)

(cherry picked from commit ea5dbcf0e6710a689acd28ef196aa978df27dcea)

Co-authored-by: Jeongseok Kang <jskang@lablup.com>

github-actions[bot] committed:
[v3-0-test] Make log redaction safer in edge case when redaction has an error (#54046) (#54048)

* [v3-0-test] Make log redaction safer in edge case when redaction has an error (#54046)
(cherry picked from commit 93ba1c5db63579c787498e40e87f2cde10143632)

* Fix bad mocking in test

Co-authored-by: Ash Berlin-Taylor <ash@apache.org>

github-actions[bot] committed:
[v3-0-test] Only forward debugger ports when debug components are used (#54051) (#54053)

When debugger option has been added to breeze in #51763 - the ports were added
to base-port.yml, but the environment variables were only set when debugging is
enabled. This hower caused warnings that the variables are not set and
defaulting them to empty string when no debug components were used.

Instead - we moved the debugger ports to separate compose file and
only use the compose file when debug components are used.
(cherry picked from commit 939700f2f3ff98f6c38958396adb9bed576707f2)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

ashb committed:
Fixup constraint generation script

This change came from #46891, but that was a much bigger change to enable
Python 3.13 which we don't want to backport

(cherry picked from commit 488744c9ada417883e223da516b7fce94b63581a)

potiuk committed:
Apply pre-commits

potiuk committed:
Run upgrade installers

potiuk committed:
Simplify installing airflow from GitHub repo in breeze (#54070) (#54076)

Breeze had the possibility of installing airflow from a branch of
any Github repo - by providing VCS url, but it's been broken since
the split of distributions. This PR adds capability of using
`owner/repo:branch` as `--use-airflow-version` and the installation
will be done using this GitHub repo.

Note! Such installation will NOT (currently) compile the assets
so you will not be able to run api_server easily after such installation
We might want to improve that in the future.

(cherry picked from commit 40ccc55536b9c081104e54e24ccf3a7de85151e5)

amoghrajesh committed:
Limit paramiko to `< 4.0.0` till we remove DSS support (#54078)

(cherry picked from commit 18d75db36c52c7a4e794b041a2fc2359aabe38ab)
